### PR TITLE
fix symbol flickering across antimeridian

### PR DESCRIPTION
**Pull request by ansis**
fix https://github.com/mapbox/mapbox-gl-native/issues/11878

This fixes:
- symbol flickering when crossing the antimeridian
- tile flickering when crossing the antimerdian

I pushed the `glfw-app` changes I used to test the flickering to [port-antimeridan-fix-debug](https://github.com/mapbox/mapbox-gl-native/tree/port-antimeridian-fix-debug)

@ChrisLoer

### CODE REVIEW

File: src/mbgl/text/cross_tile_symbol_index.cpp [hunk 1]
```diff
@@ -61,6 +61,32 @@ void TileLayerIndex::findMatches(std::vector<SymbolInstance>& symbolInstances, c
...
+ */
+void CrossTileSymbolLayerIndex::handleWrapJump(float newLng) {
+
+    const int wrapDelta = ::round((newLng - lng) / 360);
+    if (wrapDelta != 0) {
+        std::map<uint8_t, std::map<OverscaledTileID,TileLayerIndex>> newIndexes;
+        for (auto& zoomIndex : indexes) {
+            std::map<OverscaledTileID,TileLayerIndex> newZoomIndex;
+            for (auto& index : zoomIndex.second) {
+                // change the tileID's wrap and it to a new index
```
Comment (ChrisLoer) [inline src/mbgl/text/cross_tile_symbol_index.cpp]: Nit: Looks like this should be something like "and move its index"?

**Comment by ChrisLoer** [comment]
Note: I had to restart the ios-sanitize and android-debug-arm-v7 tests to get them to pass, but it looked like it was just some sort of unrelated CI timeout.

**Comment by friedbunny** [comment]
Added the `needs changelog` label here, as this should have platform changelog entries added at some point.

**Comment by friedbunny** [comment]
iOS changelog added in #12086.

**Pull request merged**